Open WebUI App for Dashboard

mik-tf opened the pull request:
### Description

- Add Open WebUI app for dashboard
- Closes threefoldtech/tfgrid-sdk-ts#3868 
- The flist uses watchtower to automatically update open webui

### Specs

- We don't need ipv4
- User can deploy with gateway domain or custom domain
- Only works for CPU now
  - Later we can do a version for GPU

### General Checklist

- [x] Build pass: yes
- [x] Documentation: https://github.com/threefoldtech/info_grid/pull/739


### Notes

- Done with the collaboration of @scottyeager

mik-tf committed:
add openwebui app for gw

mik-tf committed:
removed ipv4 option as it is not needed

mik-tf committed:
openwebui add link to manual docs

amiraabouhadid reviewed (commented):
i selected node 156, but it keeps deploying on node 313 which fails 
https://www.loom.com/share/625d1cd017cd40fda4cb3fe0d800ed75?sid=024f700b-dd09-433b-b3e9-f981ece2d183

amiraabouhadid commented on packages/playground/public/info/openwebui.md:
```diff
@@ -0,0 +1,5 @@
+title: Open WebUI
+---
+
+Open WebUI is an extensible, feature-rich, and user-friendly self-hosted AI platform designed to operate entirely offline. It supports various LLM runners like Ollama and OpenAI-compatible APIs, with built-in inference engine for RAG, making it a powerful AI deployment solution. For more details, check [Open WebUI documentation](https://www.manual.grid.tf/documentation/dashboard/solutions/openweb ...
```
the link gives a 404 page
![image](https://github.com/user-attachments/assets/6d781e19-bf7e-4e01-aea7-490881c665e2)

mik-tf replied:
Yes it is normal. The PR is not merged yet: https://github.com/threefoldtech/info_grid/pull/739

We can merge the manual PR when the dashboard PR is accepted and released.

maayarosama commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,239 @@
+        :medium="{ cpu: 2, memory: 4, disk: 100 }"
+      />
+
+      <Networks
```
why is ipv4 not added here?

mik-tf replied:
This type of deployment does not need nor benefit from IPv4. Users can

- use gateway domain + gateway network
- use custom domain + gateway network

The ipv4 would just cost more TFT for the deployment without benefits.

maayarosama replied:
Most of the other solutions uses domains too, but we give the user the option to enable ipv4

mik-tf replied:
Yes I understand this. But for this app it is not necessary nor useful to have IPv4. I discussed this with @xmonader (on Telegram) and he was OK with Open WebUI app not having IPv4.

I propose that we don't add it for now, and if users request it, we can add it in the future. Wdyt?

maayarosama commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,239 @@
+        :has-custom-domain="selectionDetails?.domain?.enabledCustomDomain"
+        require-domain
+      />
+
```
From what I can see from the issue, deploying on a GPU Node is supported. Is there a reason why the GPU switch isn't added here?

mik-tf replied:
For the first iteration of the app, we do not support GPU.

For this we need a different docker image from open webui and we need to install the proper GPU driver (AMD or NVidia). I will be able to add GPU support later but not for this PR. For now we focus on having CPU only.

maayarosama commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,239 @@
+const selectionDetails = ref<SelectionDetails>();
+
+const name = ref(generateName({ prefix: "owui" }));
+const gitUrl = ref("");
```
This variable isn't used, please remove it

mik-tf replied:
Good observation. Removed in this commit: f412d0485dbd59b3da3c2148eafbe6d56a630d37

maayarosama commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,239 @@
+
+const name = ref(generateName({ prefix: "owui" }));
+const gitUrl = ref("");
+const gitBranch = ref("");
```
This variable isn't used, please remove it

mik-tf replied:
Done with commit above f412d0485dbd59b3da3c2148eafbe6d56a630d37

maayarosama commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,239 @@
+const name = ref(generateName({ prefix: "owui" }));
+const gitUrl = ref("");
+const gitBranch = ref("");
+const root = ref("");
```
This variable isn't used, please remove it

mik-tf replied:
Done in commit: 11ce5ef65319baae72402e10f03252a3c935bd1b

maayarosama commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,239 @@
+const root = ref("");
+const domain = ref();
+
+const { ipv6, planetary, mycelium, wireguard } = useNetworks();
```
shouldn't ipv4 be added?

mik-tf replied:
See https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3869#discussion_r1950025923

mik-tf commented:
> i selected node 156, but it keeps deploying on node 313 which fails https://www.loom.com/share/625d1cd017cd40fda4cb3fe0d800ed75?sid=024f700b-dd09-433b-b3e9-f981ece2d183

Thanks for testing. I can't reproduce on my end. Deployed on node 156 devnet:

![image](https://github.com/user-attachments/assets/745993ac-81ef-45db-b886-b4f8786967c2)

As I understand the issue your encountered is with the gateway, not the deployment per se.

mik-tf committed:
removed unused variables

mik-tf committed:
removed unused variable

mik-tf commented:
@amiraabouhadid @maayarosama thanks for the review. Lmk if anything else is needed.

mik-tf committed:
changed solution code to oi from owui to be consistent

maayarosama commented:
Works fine
![Screenshot from 2025-02-10 12-09-06](https://github.com/user-attachments/assets/343a84f6-e96e-408d-a822-d0c5fd9d0d5a)

mik-tf commented:
Thanks @maayarosama for the review + approval. One more to go!

Lmk anyone if anything else is needed.

mik-tf commented:
# Update

- I started working directly on the GPU version
- The flist full vm gpu version is done and approved
- I am finalizing the dashboard app (only need to add the gateway domain section and I will make the PR as ready for review)

mik-tf committed:
new full vm gpu version of open webui app

mik-tf commented:
# Update

- We decided to publish right away the GPU version of the app, instead of CPU only
- For this we needed to use a full VM instead (the flist full vm version works 100% and is already accepted in tf-images repo)
- WIP (see below)
- Note: I will get rid of unused variables when I find the gateway fix (failed Lint test)

# TODO

- All that is left to do is to finalize the logic for the gateway domain (with port 8080) and custom domain (with port 8080). This means also adding the envs variable OPENWEBUI_DOMAIN.
- Ideally we can present this at ETH Denver. I think I can find the fix on my own, but if devs have any time to check on Monday, it would be VERY appreciated as I am sure you guys can know quickly what I am missing. @xmonader. That being said, I can find it with enough time on my side. Thanks!

mik-tf committed:
add subdomain, domain

scottyeager committed:
Fix gateway for openwebui

mik-tf committed:
update value for dialog OI

scottyeager committed:
Update openwebui deployment

mik-tf committed:
update openwebui, no ipv4, no domain button

mik-tf committed:
fix undefined var

mik-tf commented:
# Update

The PR is ready for review with GPU version.

- Scott fixed the web gateway flow.
- We confirmed with Scott that the env var was the issue, see here: https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3908
   - So for now we don't have the button Access Domain, but show the domain URL when the deployment is successful (successful message shows ${domain}
- The app works for gpu or cpu, tested on 2 nvidia nodes, e.g. node  7510
  - Note that node 7510 on mainnet shows as shared sometimes. To take into account if you test it. See here: https://github.com/threefoldtech/tfgrid-sdk-go/issues/1330

amiraabouhadid reviewed (commented):
please merge development into this branch to avoid having wireguard network only selected but mycelium only domains showing up in the domains selection
otherwise, you'll end up with this err (i think)
![image](https://github.com/user-attachments/assets/d7fd5c27-86db-4010-b08e-f3a284c07eb3)

mik-tf committed:
Merge branch 'development' into development_ai
merge dev into dev_ai

mik-tf committed:
add manage domain button to see gw domain

mik-tf commented:
@amiraabouhadid Great, I merged dev into dev_ai. 

Lmk if anything else is needed. Otherwise I think this PR can be approved. Thanks!!

mik-tf commented:
# Update

- Flist PR is accepted: https://github.com/threefoldtech/tf-images/pull/332
- info_grid PR is accepted (docs for dashboard app): https://github.com/threefoldtech/info_grid/pull/739

# TODO

- We need one more approval on this Dashboard App PR to merge it, then I'll merge the tf-images and info_grid PRs so it all works. @amiraabouhadid lmk. Thanks!

amiraabouhadid reviewed (changes requested):
was able to deploy 
![image](https://github.com/user-attachments/assets/57b7bd66-f744-453e-9b2c-f271b61932fe)
please fix subdomain name though in add new domain 
![image](https://github.com/user-attachments/assets/42e88991-2b18-41a9-8a26-8e79966290dc)
second subdomain should start with oi, not vm

mik-tf commented:
Good to know the deployment worked!

I was using the Manage Domain button from Full VM to overcome an issue where we can't use env variable with GPU (other than the SSH key), please check this issue for more info: https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3908. The fix was found by @scottyeager. So I integrated the fix and tested it. It works very well.

Now we have the proper button, as we have with the other micro VMs, see below:

![image](https://github.com/user-attachments/assets/03972a8b-9081-40ef-aac9-851feb25689d)

mik-tf committed:
fixed gpu env issue, set proper button for open webui deployment

amiraabouhadid commented:
I get this error when I try to deploy now 
![image](https://github.com/user-attachments/assets/be2c369e-2a3e-47d0-a9b0-22b59d1cee35)
I have also noticed that there is an issue with node cards loading, perhaps has to do with filtering for gpu nodes 
![image](https://github.com/user-attachments/assets/79b35c72-2a2f-42ca-96dc-ca78d0bf73a2)

mik-tf commented:
@amiraabouhadid I think you might need to rebuild the Dashboard with the whole current branch. I had the same issue as you but I just needed to rebuild the dashboard/reclone the repo. It might be the older stuff that got in the way.

I deployed 12 hours ago and it worked. And I just deployed now and it works.

![image](https://github.com/user-attachments/assets/c5850f9f-a562-45b2-bbfb-e3e5d6fdb836)

I can un-reserve node 7510 and you can give it a try. On mainnet.

0oM4R commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,263 @@
+          v-model="solution"
+        />
+
+        <Networks
```
why not adding ipv4 ?

mik-tf replied:
It is not necessary for this app. This decision was approved by Thabet.

0oM4R replied:
We should not remove the ipv4, we need to be consistent with other solutions. 
And for now we couldn't test due to the connectivity issue I mentioned here https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3869#pullrequestreview-2677663022

So please add the public IP option

mik-tf replied:
This is weird. I could test a few hours ago and it worked, using node ID 320 on devnet. Doing tests now too on node ID 320 and it works 100%. With 2x Nvidia 4090 GPUs actually.

For the ipv4, it's not necessary and would need more dev-time. The idea is that we can use mycelium and/or web gateways, no need for ipv4. As I said, it had been approved as such previously by Thabet. What I propose is we accept and merge to dashboard now. and we can do a later update on the app with ipv4.

For now it works 100% as is. You can see below with nvidia-smi command, both gpus are used, and deepseek-r1 70b parameters is fast and effective.

![image](https://github.com/user-attachments/assets/7e3455cb-02f7-4619-a224-719bf8799da1)
![image](https://github.com/user-attachments/assets/9ae02c9b-29e8-4644-b98b-10f659bc43fe)

mik-tf replied:
Resolving this as discussed we won't focus on IPv4 for now. Also, note that the nodes with great GPUs often or always don't have IPv4 anyway, so users will need to use the gateways anyway. 

Note that we are having discussions with partners to bring more nvidia 4090 nodes on the grid, and we want to have that app on the Dashboard mainnet asap to showcase. Currently, the app works well for the intended purpose. 

Feel free to discuss/give feedback if you think it's needed.

0oM4R replied:
how did you deploy on node 320, full VM with the same flist right? but we need to deploy from the application you added to verify node 320 is zoslight and it is not supporting wireguard.

I still think we should add ipv4 as the rest of the solutions @xmonader @AhmedHanafy725

mik-tf replied:
Gotcha. Will show the current setup to partners in a call in a few hours, but after this, I will re-deploy with the app itself and lyk the results. Thanks again for all the work around this app.

mik-tf replied:
Here's what I have 

![image](https://github.com/user-attachments/assets/4a43661d-8c55-4c0b-aeae-09d2cd3c4819)


What do you propose for the next steps? @0oM4R Thanks.

mik-tf replied:
@0oM4R Hi can you give more info on what you want next for this PR? See picture above with node 320 not supporting wireguard.

If you do want to go with IPv4, can you have a look at the code here and propose a way forward? Thanks.

mik-tf replied:
Added IPv4

mik-tf replied:
IPv4 added. Setting as resolved.

0oM4R commented on packages/grid_client/src/zos/zmachine.ts:
is that related to the pr? i think the fix already merged

mik-tf replied:
Exactly. The issue waa found by Scott and I while doing this app. I will merge dev branch to integrate it officially.

0oM4R commented on packages/grid_client/src/zos/zmachine_light.ts:
This one to; I think it is not related; please make sure to pull the development branch into this branch

mik-tf replied:
Yes. As above. Will do.

mik-tf replied:
Done

0oM4R commented on packages/playground/src/views/solutions_view.vue:
```diff
@@ -99,6 +99,13 @@ export default {
+          "Open WebUI is an extensible, feature-rich, and user-friendly self-hosted AI platform designed to operate entirely offline.",
+        icon: "openwebui.png",
+        route: DashboardRoutes.Applications.Openwebui,
+      },
```
make sure to open an issue to include the release date

mik-tf replied:
Issue: https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3975

0oM4R commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,263 @@
+const domain = ref();
+const flist = ref<Flist>({
+  name: "Ubuntu-24.04 Open WebUI Instance",
+  value: "https://hub.grid.tf/idrnd.3bot/ubuntu-24.04_fullvm_oi.flist",
```
The flist should be an official one. Please check with ops to include it in the official lists like the other solutions.

mik-tf replied:
Good. Yes the flist got approved in tf-images. Will do this thanks.

mik-tf replied:
Issue created here: https://github.com/threefoldtech/tf-images/issues/345
Will update this development_ai branch with new link when it's up. Thanks.

mik-tf replied:
Done here: https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3869/commits/8ef9ed2d2386a0318b258e188e072a386fadd2d6

@0oM4R should be good to go now, everything is updated as requested. Lmk if there's anything else. Thanks.

0oM4R commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,263 @@
+            "
+        >
+          <v-switch color="primary" inset label="GPU" v-model="hasGPU" hide-details />
+        </input-tooltip>
+        <input-tooltip inline tooltip="Click to know more about dedicated machines." :href="manual.dedicated_machines">
+          <v-switch color="primary" inset label="Dedicated" v-model="dedicated" hide-details />
+        </input-tooltip>
+
```
those toggles got changed on https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3837 
please make sure to update them

mik-tf replied:
Good. Will do.

mik-tf replied:
Done here: https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3869/commits/892fefd05a6ead64df8b9b869ee55deb1c1abef7

0oM4R reviewed (changes requested):
Great solution! Thank you for your contribution @mik-tf 💻

0oM4R commented:
I tried to run the branch on mainnet and deployed on a GPU node
after waiting I still have a bad gateway I ssh to the machine to get the container logs but I didn't found any running container 
![image](https://github.com/user-attachments/assets/765280fe-0203-4f53-a65a-b255ace2f2ca)

a found script `start-containers.sh` ran it and it started to pull the images!
![image](https://github.com/user-attachments/assets/df153421-f86f-4b87-b7f5-724ccaedc10f)

any idea what I did wrong?

deployment json:

```json
{
  "version": 0,
  "type": "zmachine",
  "contractId": 892261,
  "nodeId": 7343,
  "name": "oiecmzu",
  "created": 1740574131,
  "status": "ok",
  "message": "",
  "flist": "https://hub.grid.tf/idrnd.3bot/ubuntu-24.04_fullvm_oi.flist",
  "publicIP": null,
  "myceliumIP": "5da:f0e9:74db:4e5a:ff0f:0:400:2",
  "interfaces": [
    {
      "network": "nwd4cc4",
      "ip": "10.20.4.2"
    }
  ],
  "capacity": {
    "cpu": 5,
    "memory": 24890
  },
  "mounts": [],
  "env": {
    "SSH_KEY": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDKkaopr/tRilmLprntgqAO6JxnhzhrrB02KeUsNys9qj/G4RtXK4hKZo3yj42Kuoub53TxoW/BfZSRUbY0VNUgZSsifCyDE4g1UXi83ic+uepPb1VIfzlFdtZPUo/dEtjfS5FM7GVAPZCDik08w2+uXeZiAKavLTDQFh2cuIqE5QnQ44enODTMkSDPVJSKJ6NSoNcrY2I++AtcxgNFzy/7YWoT/bA19CliGqxBMSQ/GjOEAF3iQjUPc5LYcqCYAc0Y6WPt2l8uEVvhJAtsLelGApt8v/Nq/OBEKJpUQB2cfkyKlwLLphKFtQ8gcKjrbpE47lVsfbW68uqCw/5avA71HlS6AXxMeda8GZ99UqDTOLoUbd+EEo17hiHs6Nvle8DHNfWdGT2wx+DhzZeCO719UadmQxFLYDd75dDH5gLkMxr9JXPWDqdxvsyMZilxZPk3UQbK811obYSMrc9L+u8vwLs0weUBxnygutnU0eF9cRQxFgx3zOOfwI1ugI9SgSU= omarkassem099@gmail.com",
    "OPENWEBUI_DOMAIN": "oi8574oiecmzu.gent01.grid.tf"
  },
  "entrypoint": "",
  "metadata": "",
  "description": "",
  "rootfs_size": 2684354560000,
  "corex": false,
  "gpu": [
    "0000:02:00.0/1002/554d"
  ],
  "deploymentName": "oiecmzu",
  "projectName": "openwebui/oiecmzu",
  "billing": "No Data Available",
  "wireguard": "[Interface]\nAddress = 100.64.20.3/32\nPrivateKey = kk4UxHaWAWUhrZmtdmqMpxBgB4IhXZ1mwWlVoP62ZUQ=\n\n[Peer]\nPublicKey = vi+wvrFKtDjC4gC3Xu+HtKzHZcRpKuCP9qKLIAkHJjM=\nAllowedIPs = 10.20.0.0/16, 100.64.20.0/32\nPersistentKeepalive = 25\nEndpoint = 185.69.166.140:26276"
}
```

mik-tf commented:
@0oM4R you did nothing wrong. It just takes time for the deployment to finish. We put a basic message in the successful message. When it is done the bad gateway disappears and you see the OI page.

Thanks for testing.

mik-tf commented:
The app deploys the scripts automatically with systemd services gpu-setup and start-containers. If there is a gpu it installs the driver and if not OI is run with CPU only.

amiraabouhadid commented:
I deployed more than 15 mins ago on mainnet and it still gives bad gateway
![image](https://github.com/user-attachments/assets/83af3e3c-a0a7-4112-bac3-9b9c21b92fa6)
![image](https://github.com/user-attachments/assets/d9ab4473-fda1-475e-8185-ad502791420e)
how long do I need to wait ?

mik-tf committed:
Merge remote-tracking branch 'origin/development' into development_ai

mik-tf committed:
fix conflict in router/routes

mik-tf commented:
@amiraabouhadid sorry for the late response. It can take 30 min to deploy. It's very long as it needs to install the GPU driver and pull the docker. But once it's done, the docker containers are automatically updated via Watchtower, so it was a trade-off vs having just the open webui pre-seeded on the flist with a fixed version.

mik-tf committed:
add rented by me for OI app

mik-tf committed:
add promoted flist for OI app

mik-tf commented:
Just to it's clear in the PR: this should be good to go. Resolved all the changes requested. Thanks.

amiraabouhadid commented:
deployment was indeed successful after 30 mins but how do I go about selecting a model?
![image](https://github.com/user-attachments/assets/9ade8072-c918-4888-b9d4-fa272e17ef34)
I've looked at the docs to find out how but there wasn't much about how to run it after successful deployment

mik-tf commented:
@amiraabouhadid for this I would consider that it is up to users to check docs from Open WebUI and Ollama. But I can add some basic info so users can get started, good suggestion.

In short:  To add Ollama models to Open WebUI, go to the Open WebUI interface and select "Settings" » "Models". From there, you can download and manage your models, e.g. "deepseek-r1".

mik-tf commented:
@amiraabouhadid added some basic info here: https://github.com/threefoldtech/info_grid/pull/761
For the rest users should check OI docs which I linked in the manual. Thanks.

0oM4R commented on packages/playground/src/router/routes.ts:
```diff
@@ -62,6 +62,7 @@ enum ApplicationRoutes {
   Discourse = "/deploy/labs/discourse/",
   Taiga = "/deploy/labs/taiga/",
   Owncloud = "/deploy/labs/owncloud/",
+  Openwebui = "/deploy/applications/openwebui/",
```
Please update the route
```suggestion
  Openwebui = "/deploy/laps/openwebui/",
```

0oM4R commented on packages/playground/src/router/index.ts:
```diff
@@ -188,6 +188,28 @@ function createApplicationsRoutes(): RouteRecordRaw[] {
+          path: [
+            { title: "Deploy" },
+            {
+              title: "Applications",
```
```suggestion
              title: "Labs",
```

0oM4R commented on packages/playground/src/weblets/tf_openwebui.vue:
please make sure to update the toggles with the new flow on development 
check: https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3952

mik-tf replied:
As I understand, this is already updated.

- See https://github.com/threefoldtech/tfgrid-sdk-ts/blob/7d13aa7174c871bbb4cc47b6675e1531bc9726af/packages/playground/src/weblets/tf_openwebui.vue#L59
- See https://github.com/threefoldtech/tfgrid-sdk-ts/blob/7d13aa7174c871bbb4cc47b6675e1531bc9726af/packages/playground/src/weblets/tf_openwebui.vue#L62

I tested it and it has the same effect as on other apps.

0oM4R replied:
the missing part is to toggle on the rented by me on toggling on GPU 

https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3952/files#diff-27f21c1ae9d2d7befc0140dc00e3877a604ad5b912d9e505b828c9e7b6d0842eR225-R242

mik-tf replied:
Gotcha. Thanks for the clarification. Totally missed it. Did it here: https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3869/commits/da19589859d21e8e504f03113ba893c147b60b37

mik-tf replied:
Resolving as it's done.

0oM4R reviewed (changes requested):
I couldn't test. I can't find nodes on the dev net, and it looks like nodes on the main net have a connection issue. 
I tried without GPU cards on devnet, and it works fine. 
<img width="1501" alt="image" src="https://github.com/user-attachments/assets/d9a4bb3b-a751-4420-815a-c353b3c74045" />

mik-tf committed:
Update packages/playground/src/router/routes.ts

Co-authored-by: Omar Kassem <omarksm09@gmail.com>

mik-tf committed:
Update packages/playground/src/router/index.ts

Co-authored-by: Omar Kassem <omarksm09@gmail.com>

mik-tf commented:
> I couldn't test. I can't find nodes on the dev net, and it looks like nodes on the main net have a connection issue. I tried without GPU cards on devnet, and it works fine. 

You can simply build the Dashboard locally and change the config files to get unto mainnet, there you can test on many different GPUs. 

Basically, build the config.js with the script build-env.sh, the replace this new file with the config.js in public directory. Then you get the development dashboard on mainnet and you can test GPUs.

![image](https://github.com/user-attachments/assets/c59d59bf-45f1-45f1-aa00-414ee736e398)


Lmk if you need more info for this. Thanks for the tests and feedback.

0oM4R commented:
> > I couldn't test. I can't find nodes on the dev net, and it looks like nodes on the main net have a connection issue. I tried without GPU cards on devnet, and it works fine.
> 
> You can simply build the Dashboard locally and change the config files to get unto mainnet, there you can test on many different GPUs.
> 
> Basically, build the config.js with the script build-env.sh, the replace this new file with the config.js in public directory. Then you get the development dashboard on mainnet and you can test GPUs.
> 
> ![image](https://private-user-images.githubusercontent.com/77026219/421909061-c59d59bf-45f1-45f1-aa00-414ee736e398.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE3ODY3MDksIm5iZiI6MTc0MTc4NjQwOSwicGF0aCI6Ii83NzAyNjIxOS80MjE5MDkwNjEtYzU5ZDU5YmYtNDVmMS00NWYxLWFhMDAtNDE0ZWU3MzZlMzk4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEyVDEzMzMyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFlMWE0MDY0MTExM2E3NTk4Y2U1MGQzM2M3NzFmMjQwYzU4YmQ1MTRjNjBmYWY3NjY3NjVkM2E0ZDJmNTdlMWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.F8hxIpo-qgSoQbldSiE3W8Tj5uGrWTu0276N4anCvm0)
> 
> Lmk if you need more info for this. Thanks for the tests and feedback.

I did, but I couldn't access the machine, and it gave me a bad gateway for more than 1 hour, i see its not related to your pr its a connection issue; i just had to mention this point in my review.

mik-tf commented:
> I did, but I couldn't access the machine, and it gave me a bad gateway for more than 1 hour, i see its not related to your pr its a connection issue; i just had to mention this point in my review.

Gotcha. Thanks for the clarification.

amiraabouhadid commented:
thanks for adding to the docs! I indeed got the resources err because the node I was using had only CPU 
![image](https://github.com/user-attachments/assets/eaee49b5-5243-4200-93b1-46ebed6ccd92)
will try with gpu , but perhaps we should make it a requirement that a node has GPU in order to deploy an instance

amiraabouhadid commented on packages/playground/src/router/routes.ts:
```diff
@@ -62,6 +62,7 @@ enum ApplicationRoutes {
   Discourse = "/deploy/labs/discourse/",
   Taiga = "/deploy/labs/taiga/",
   Owncloud = "/deploy/labs/owncloud/",
+  Openwebui = "/deploy/laps/openwebui/",
```
```suggestion
  Openwebui = "/deploy/labs/openwebui/",
```

amiraabouhadid commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,270 @@
+        >
+          <v-switch color="primary" inset label="GPU" v-model="hasGPU" hide-details />
+        </input-tooltip>
+        <!-- <input-tooltip inline tooltip="" :href="manual"> -->
```
please remove unnecessary commented out code

mik-tf replied:
Good suggestion. Done.

mik-tf committed:
Update packages/playground/src/router/routes.ts

Co-authored-by: Amira <56790126+amiraabouhadid@users.noreply.github.com>

mik-tf committed:
removed comments

mik-tf commented:
> thanks for adding to the docs! I indeed got the resources err because the node I was using had only CPU

It works with CPU people just need to put more CPU resources. It can be used to deploy basic low requirements models. I would keep it so it is more versatile.

mik-tf committed:
removed unnecessary comments

mik-tf committed:
updated specs for oi vm

mik-tf commented:
@amiraabouhadid based on your feedback, I added a section on the manual with proper info on using gpu vs cpu-only:

![image](https://github.com/user-attachments/assets/89f7e69b-39e4-4293-8c4c-690b10c02cf5)

On the app itself, I set the small, medium, large to 

![image](https://github.com/user-attachments/assets/cb7f46af-ecb5-4694-b713-150b88ea9a36)

and updated the docs accordingly.

So now it is very clear for users how to use the app for gpu and cpu-only versions. Thanks.

mik-tf commented:
@amiraabouhadid @0oM4R as I understand, all requested changes have been resolved. Please lmk if anything else is needed, otherwise I think we can merge this. Thanks.

0oM4R commented:
@mik-tf please check https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3869#discussion_r1999094670

mik-tf committed:
openwebui adjust toggle buttons with gpu and rented by me

amiraabouhadid commented:
hey @mik-tf thanks for updating the docs and fixing the GPU toggle, I'm just waiting to get some tokens on mainnet so I can further test this on a node with GPU. Thanks for your hard work and patience.

mik-tf commented:
Of course! Curious to know how it goes now with updates. Lmk. Thanks.

mik-tf committed:
Merge branch 'development' into development_ai
merging dev

mik-tf committed:
set ipv4 for OI app

mik-tf commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -228,13 +232,19 @@ async function deploy() {
     await deployGatewayName(grid, selectionDetails.value?.domain, {
       subdomain,
       ip: vm[0].interfaces[0].ip,
-      port: 8080,
```
Changed port from 8080 to 80 so custom domain works. Will update the flist too for this. Now this [flist](https://hub.grid.tf/idrnd.3bot/ubuntu-24.04_fullvm_oi.flist) has the port 80, will ask to set to official tf image list.

mik-tf replied:
Flist updated. Can be tested.

mik-tf commented:
@0oM4R should be good to test with ipv4! Thanks

mik-tf commented:
Now the flist+dashboard app are up to date with the latest changes, with IPv4 access. Should be good for final tests @0oM4R! 

Ideally we can merge this and push to mainnet so users can test the Nvidia 4090 GPU nodes from our partners that will be on mainnet with 3.16 release (and gpu bug fixes). Thanks!

mik-tf commented:
@0oM4R can we merge this? Everything is OK. Thanks!

0oM4R commented:
hello @mik-tf, I was off for the last 2 weeks, sorry for being late.
all good now but I can't test the solution with a GPU node as the mentioned issue https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3869#issuecomment-2717912263

mik-tf commented:
Thanks for the info @0oM4R. All good. I hope you had a nice 2 week off :D

As I understand the PR can be approved if everything works, as the issue you mention is not related to this PR.

mik-tf commented:
I see here we already have 2 approvals. Also there are 2 requested changes by @0oM4R  and @amiraabouhadid. I resolved all requested changes. 

Perhaps you two can set the requested changes as done, as I think I covered everything. If not, lmk what to change and I can update the PR so we can clear the 2 requested changes.

xmonader commented:
@0oM4R if you managed to run it at least once, merge. and any issue that comes up, @mik-tf will handle it

0oM4R commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,281 @@
+    layout.value.setStatus("failed", normalizeError(e, "Failed to deploy an Open WebUI instance."));
+  }
+
+  if (selectionDetails.value!.domain!.enabledCustomDomain && ipv4.value) {
```
```suggestion
  if (!selectionDetails.value?.domain?.enableSelectedDomain) {
```

this is shorter and consist with other solutions

0oM4R commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,281 @@
+  layout.value.reloadDeploymentsList();
+  layout.value.setStatus(
+    "success",
+    `Successfully deployed an Open WebUI instance. The domain URL is "${domain}". You might need to wait a couple of minutes for the installation to complete, e.g. if you see Bad Gateway when opening the webpage, simply wait and refresh the page.`,
```
i don't think there is a need to show the URL here, as other solutions show the URL in the deployment details.

mik-tf replied:
https://github.com/threefoldtech/tfgrid-sdk-ts/commit/070975ad8174f192caf20c589b556d8d9adfb952

mik-tf replied:
Good idea.

0oM4R commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,281 @@
+const solution = ref() as Ref<SolutionFlavor>;
+const selectedSSHKeys = ref("");
+const name = ref(generateName({ prefix: "oi" }));
+const domain = ref();
```
please remove this as it is not used

mik-tf replied:
https://github.com/threefoldtech/tfgrid-sdk-ts/commit/eaf255be54849d10ff5af0a96fa2ca440be84ff7

Good point. Lmk if it's OK I removed 3 variables unused. Thanks.

0oM4R commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,281 @@
+
+const layout = useLayout();
+const profileManager = useProfileManager();
+const tabs = ref();
```
please remove this as it is not used

0oM4R commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,281 @@
+import { type Flist, ProjectName } from "../types";
+import { deployVM, type Disk } from "../utils/deploy_vm";
+import { deployGatewayName, getSubdomain, rollbackDeployment } from "../utils/gateway";
+import { getGrid } from "../utils/grid";
```
please remove this as it is not used

0oM4R reviewed (commented):
I tested it on the main net, and it works after some time. The functionality is good now, just a few code enhancements, and we will be ready to go. Thank you for your great effort

mik-tf committed:
Update packages/playground/src/weblets/tf_openwebui.vue

Co-authored-by: Omar Kassem <omarksm09@gmail.com>

mik-tf committed:
removed domain url in success answer

mik-tf committed:
removed unused variables

mik-tf commented:
Thanks a lot for the review once more.

Here's the 3 lines removed: https://github.com/threefoldtech/tfgrid-sdk-ts/commit/eaf255be54849d10ff5af0a96fa2ca440be84ff7
And the domain URL has been removed in the success message.

Lmk if there's anything else!

0oM4R commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,278 @@
+  { immediate: true },
+);
+
+function finalize(deployment: any, domain: string) {
```
its not using the domain now, please remove the domain here and in the line this function got called

0oM4R replied:
@mik-tf small change please make sure to address it before merge

mik-tf replied:
Very good call. Thanks. 

Done: https://github.com/threefoldtech/tfgrid-sdk-ts/pull/3869/commits/8308863c0ca5267d8b2006a20db6c47e0b95d3db

mik-tf committed:
OI app removed unnecessary domain calling

mik-tf commented:
@amiraabouhadid can you confirm the changes have been done? Thanks! (1 requested change).

amiraabouhadid commented on packages/playground/src/weblets/tf_openwebui.vue:
```diff
@@ -0,0 +1,278 @@
+  layout.value.reloadDeploymentsList();
+  layout.value.setStatus(
+    "success",
+    `Successfully deployed an Open WebUI instance. You might need to wait a couple of minutes for the installation to complete, e.g. if you see Bad Gateway when opening the webpage, simply wait and refresh the page.`,
```
```suggestion
    `Successfully deployed an Open WebUI instance. Please keep in mind that the installation may take a few minutes to finish. If you encounter a "Bad Gateway" message while accessing the webpage, just wait a moment and refresh the page.`,
```

amiraabouhadid reviewed (changes requested):
Hey @mik-tf  thanks for your hard work. I deployed two instances on mainnet, yet no deployments are shown in the deployments list :/
<img width="926" alt="Screenshot 2025-04-07 at 20 07 50" src="https://github.com/user-attachments/assets/3803faf1-67c8-4521-a1a7-f88603468ffe" />
<img width="926" alt="Screenshot 2025-04-07 at 20 07 27" src="https://github.com/user-attachments/assets/5fc48f3e-0eca-4c7b-97dc-04d6ecf504c9" />
<img width="961" alt="Screenshot 2025-04-07 at 20 07 13" src="https://github.com/user-attachments/assets/71fadc2b-9749-4f6c-b676-c54829c4cca8" />
contracts have been created though 
<img width="926" alt="Screenshot 2025-04-07 at 20 10 21" src="https://github.com/user-attachments/assets/9732a265-1e83-43da-a859-22dec0741dd3" />

please fix the deployment list

mik-tf committed:
Update packages/playground/src/weblets/tf_openwebui.vue

Co-authored-by: Amira <56790126+amiraabouhadid@users.noreply.github.com>

xmonader commented:
Don't merge until the listing of the instances happen.

mik-tf commented:
@amiraabouhadid thanks for the tests.

I deployed on devnet, mainnet, with the branch as is, starting from scratch (clone tfgrid-sdk-ts, change to development_ai branch, deploy local dashboard), and I tested all scenarios (wireguard, mycelium, ipv4, custom domain, gateway domain), and the deployment list always works. So I cannot reproduce. Can you tell me how you deploy? Or can you start from fresh and see? Thanks.

# Dashboard OI Notes

## Devnet - development_ai branch

- deploy wireguard + mycelium, gateway domain, no custom domain,
  - instances show in deployment

![image](https://github.com/user-attachments/assets/f53d8afe-ae33-43a5-8191-fd704ac2ca28)


- deploy wireguard + mycelium + ipv4, gateway domain, no custom domain,
  - instances show in deployment

![image-1](https://github.com/user-attachments/assets/4b7d457f-81c5-4523-ab2b-ba645af922e3)

- deploy wireguard + mycelium + ipv4, gateway domain, with custom domain,
  - instances show in deployment

![image-2](https://github.com/user-attachments/assets/146e8582-a994-4ef8-8b01-2acfc176aa00)


## mainnet - development_ai branch

- deploy wireguard + mycelium, gateway domain, no custom domain,
  - instances show in deployment
![image-3](https://github.com/user-attachments/assets/3b9ea2f1-bd39-48e1-8817-4e4b66af50e4)


- deploy wireguard + mycelium + ipv4, gateway domain, no custom domain,
  - instances show in deployment
![image-4](https://github.com/user-attachments/assets/f6b8841e-bce3-4e7f-99f9-4b6846b0f908)

0oM4R commented:
listing is working fine with me against main net

https://github.com/user-attachments/assets/534d6854-1fd9-43dd-a044-783657c5ff71

amiraabouhadid commented:
hey @mik-tf I deployed on nodes with GPU on mainnet twice, both deployments weren't shown in deployment list 
https://www.loom.com/share/19dcf48f882a4fd9af04cecd30ad61d9?sid=7b7f30cb-ac4b-4809-89e0-632ce5af2f1d
please disregard it saying devnet at the top, this is a bug that we're currently looking into but this is on mainnet as indicative by the logs shown in the video and the balance

mik-tf commented:
Hmmm. That is weird. I just deployed on a node with GPU, and it shows:

![image](https://github.com/user-attachments/assets/b3567e93-7ea6-4be3-99fd-dfd9824fd907)

I would suggest to start from scratch (new tfgrid-sdk-ts repo cloned). Perhaps there's something off with your version of the Dashboard.

I can see also that @0oM4R couldn't reproduce the issue (instance not showing in deployment table).

amiraabouhadid reviewed (changes requested):
Thanks for the suggestion, but that wasn't the issue. When I try to deploy an instance on large capacity with mycelium and wireguard I get this error and it just keeps on loading endlessly 
![Screenshot 2025-04-10 at 11 08 18](https://github.com/user-attachments/assets/b0c08a9f-1cd7-4e24-bebf-a01f46e7ed25)
here are the logs for your reference 
```Lock acquired
hook.js:608 2025-04-10 11:07:12        API/INIT: Not decorating runtime apis without matching versions: Metadata/2 (1 known), TransactionPaymentApi/4 (1 known)
overrideMethod @ hook.js:608Understand this warningAI
@sentry_vue.js?v=fb230b5e:851 phase: {"applyExtrinsic":6}, section: balances, method: Withdraw, data: ["5DUR8uwERLDdUVkyjjUWrQSgCeUafAKdJLZziNcaCszbqTGJ",10112]
@sentry_vue.js?v=fb230b5e:851 phase: {"applyExtrinsic":6}, section: smartContractModule, method: ContractCreated, data: [{"version":4,"state":{"created":null},"contractId":1047889,"twinId":14108,"contractType":{"rentContract":{"nodeId":1255}},"solutionProviderId":null}]
@sentry_vue.js?v=fb230b5e:851 phase: {"applyExtrinsic":6}, section: balances, method: Deposit, data: ["5FqHp29vhYLcAyou8nNGY9T4S8bPTyMQq5GwqkvXBgb81Qjs",10112]
@sentry_vue.js?v=fb230b5e:851 phase: {"applyExtrinsic":6}, section: transactionPayment, method: TransactionFeePaid, data: ["5DUR8uwERLDdUVkyjjUWrQSgCeUafAKdJLZziNcaCszbqTGJ",10112,0]
@sentry_vue.js?v=fb230b5e:851 phase: {"applyExtrinsic":6}, section: system, method: ExtrinsicSuccess, data: [{"weight":{"refTime":884781000,"proofSize":4241},"class":"Normal","paysFee":"Yes"}]
@sentry_vue.js?v=fb230b5e:851 Lock released
@sentry_vue.js?v=fb230b5e:851 Rent contract with id: 1047889 has been created.
hook.js:608 2025-04-10 11:07:24        API/INIT: Not decorating runtime apis without matching versions: Metadata/2 (1 known), TransactionPaymentApi/4 (1 known)
overrideMethod @ hook.js:608Understand this warningAI
@sentry_vue.js?v=fb230b5e:851 Start creating the machine deployment with name oigz2la
@sentry_vue.js?v=fb230b5e:851 Start deleting the machine deployment with name oigz2la
hook.js:608 [Uncaught Error]
- Message: Failed to delete deployment with name "oigz2la".
- Stack: Error: Failed to delete deployment with name "oigz2la".
    at rollbackDeployment (http://localhost:5173/src/utils/gateway.ts:46:11)
    at async deploy (http://localhost:5173/src/weblets/tf_openwebui.vue:139:9)
- Constructor: Error
overrideMethod @ hook.js:608Understand this errorAI
hook.js:608 [Vue warn]: Error in native event handler: "Error: Failed to delete deployment with name "oigz2la"."

found in

---> <VBtn>... (2 recursive calls)
       <WebletLayout> at /Users/amira/Documents/cs/tfgrid-sdk-ts/packages/playground/src/components/weblet_layout.vue
         <AsyncComponentWrapper>
           <TfOpenwebui> at /Users/amira/Documents/cs/tfgrid-sdk-ts/packages/playground/src/weblets/tf_openwebui.vue
             <ViewLayout> at /Users/amira/Documents/cs/tfgrid-sdk-ts/packages/playground/src/components/view_layout.vue... (5 recursive calls)
               <DeploymentListManager> at /Users/amira/Documents/cs/tfgrid-sdk-ts/packages/playground/src/components/deployment_list_manager.vue
                 <VMain>... (2 recursive calls)
                   <App> at /Users/amira/Documents/cs/tfgrid-sdk-ts/packages/playground/src/App.vue
                     <AsyncComponentWrapper>
overrideMethod @ hook.js:608Understand this errorAI
:5173/#/deploy/labs/openwebui/:1 Access to fetch at 'https://dev.sentry.grid.tf/api/2/envelope/?sentry_key=b9af6796f176d1f02837a06f0da3caee&sentry_version=7&sentry_client=sentry.javascript.vue%2F8.19.0' from origin 'http://localhost:5173' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled. ```

mik-tf commented:
OK I see. I can try on my end. What is the exact specs you used? So I can try the same setup. Thanks!

mik-tf commented:
I can't test on node 1255 as it's rented.

But it looks way overprovisioned, maybe that is the reason why you have an issue!

![image](https://github.com/user-attachments/assets/341ae0aa-0eec-4ece-b748-408e49ef11fd)

@amiraabouhadid 

This most certainly looks like something is wrong with this node. Never saw this for core reservation.

amiraabouhadid commented:
node 1255 is unrented now

mik-tf commented:
Good. Though for now we can't deploy on the grid until the RMB relay fix.

![image](https://github.com/user-attachments/assets/0621b9fa-82c2-4102-a0cb-78ca817258f4)

scottyeager commented:
Some notes from my experience testing this today:

* I encountered "Failed to delete deployment" once, not sure how to reproduce
* Deployment not showing up in deployment list
    *  Seems to be node dependent, perhaps (I did not encounter this issue deploying to nodes 2925 and 5600 with the GPU enabled)
    * I can reproduce this on node 7520
    * Seems the workload isn't present in the RMB response from the node:
![image](https://github.com/user-attachments/assets/fb1bf251-5312-4181-b5f8-5f90dbc6ce78)

mik-tf commented:
Thanks for the report Scott. Tried 7520 and same as you. Strange as it worked before on that node. Perhaps something changed on RMB or dev branch sides

@0oM4R @xmonader any idea? Can this be related to the RMB or latest updates in dev branch for FullVm/GPU? Will check on my end too.

mik-tf commented:
# Update

- Very interesting data point (imo): the issue of the instance info not showing in the deployment list does not seem to be directly related to this current PR. Let me explain:

- Test Setup:
  - Current development branch with hard reset (not development_ai branch)
  - Mainnet config to access node 7520
  - Deploy a Full VM with Ubuntu 24.04, with mycelium and wireguard, node 7520 with 1x GPU, large instance capacity
  - The deployment succeeds and I see the Details/JSON pop-up window
  - But the deployment is not shown in the full vm deployment list: it says: No full vm instance found etc.
    - Sometimes it shows for a couple of seconds, then disappears.

  > Note: this does not happen if I select as above, but without wireguard
- Assessment
  - Thus it seems the issue is with GPU + wireguard on certain nodes?
  - This means it's an issue in itself, and it is not related to this PR (the Test Setup does not use the OI app nor the flist and still shows the same issue as highlighted by @amiraabouhadid)

Good catch Amira. I'll create a stand-alone issue as explained above.

mik-tf commented:
@ramezsaeed I see that you accepted the issue #4060, so do we agree with what I wrote above:That the issue is not related directly to this OI app PR? Thus we could merge this OI app PR.

Thanks!

ramezsaeed commented:
> @ramezsaeed I see that you accepted the issue #4060, so do we agree with what I wrote above:That the issue is not related directly to this OI app PR? Thus we could merge this OI app PR.
> 
> Thanks!

yes lets fix it there in the other issue #4060 and we can merge this one

ramezsaeed commented:
Test scenarios to be added:
 - test all scenarios (wireguard, mycelium, ipv4, custom domain, gateway domain),
 - deploy multiple WebUI app and make sure all shown in deployment list.
 - test delet instanse from deployment lit and make sure all contracts deleted from contract page.
 - try to deploy an instance on large capacity with mycelium and wireguard

xmonader commented:
@mik-tf once you tested the scenarios asked by Ramez, the PR is good to merge.

mik-tf committed:
Merge branch 'development' into development_ai
update with development

mik-tf commented:
# Update

- [ ] test all scenarios (wireguard, mycelium, ipv4, custom domain, gateway domain)
  - TODO: Set port 8080, and set caddy reverse proxy for HTTPS when custom domain + ipv4
  - All other scenarios work
- [x] deploy multiple WebUI app and make sure all shown in deployment list.
- [x] delete instance from deployment list and make sure all contracts deleted from contract page
- [x] try to deploy an instance on large capacity with mycelium and wireguard

mik-tf committed:
set port to 8080

mik-tf commented:
# Update

- [x] test all scenarios (wireguard, mycelium, ipv4, custom domain, gateway domain)
- [x] deploy multiple WebUI app and make sure all shown in deployment list.
- [x] delete instance from deployment list and make sure all contracts deleted from contract page
- [x] try to deploy an instance on large capacity with mycelium and wireguard

# Notes

- What still the doesn't work is the issue reported here: https://github.com/threefoldtech/tfgrid-sdk-ts/issues/4060
  - But it concerns the full VM and GPU in general, not this PR in particular.
    - This means GPU + Wireguard on e.g. 7520 mainnet is not working as reported in the issue.
  
  # Flist Update

- The Flist is being updated here: https://github.com/threefoldtech/tf-images/issues/354

xmonader commented:
Merged. It's expected that you give continuous support to it @mik-tf

mik-tf commented:
Thank you for merging. @xmonader I'll do my best to help.
